src/main-app.js

```diff
@@ -157,6 +212,36 @@ export class MainApplication extends LitElement {
       if (ev.detail.reason === "action") {
         window.location.reload();
       }
+      this._snackbar.labelText = "";
+      this._actionBtn.textContent = "";
+    });
+
+    this._store.addEventListener('change', ev => {
+      this._hideScanBg = true;
+      this.requestUpdate();
+    }, { once: true });
+
+    try {
+      this._reader = new NDEFReader();
+      this._reader.addEventListener("reading", ev => {
+        const decoder = new TextDecoder();
+        for (let record of ev.message.records) {
+          const data = JSON.parse(decoder.decode(record.data));
+          if (data.product) {
+            this._store.set(data.product, data.description);
+          }
+        }
+      });
+    } catch(err) {
+      console.error("Reading NFC tags is not supported");
+    }
+
+    this._btnScan.addEventListener('click', async ev => {
```

Comment:
no need to add async as you don't use await inside.

Also you can probably easier add these in the html code `@click="this._startScan"

Reply:
Done.